contentcuration/contentcuration/assets/channelList/views/ChannelSet/ChannelSetModal.vue

```diff
@@ -0,0 +1,354 @@
+        return this.$tr('editingSetHeader');
+      },
+      open: {
+        get() {
+          return true;
+        },
+        set(open) {
+          if (!open) {
+            this.close();
+          };
+        },
+      },
+      name: {
+        get() {
+          return this.channelSet.name || '';
+        },
+        set(value) {
+          this.updateChannelSet({ id: this.channelSetId, name: value });
+        },
+      },
+      description: {
+        get() {
+          return this.channelSet.description || '';
+        },
+        set(value) {
+          this.updateChannelSet({ id: this.channelSetId, description: value });
+        },
+      },
+      channels: {
+        get() {
+          return this.channelSet.channels || [];
+        },
+        set(value) {
+          this.updateChannelSet({ id: this.channelSetId, channels: value });
+        },
+      },
+      enableSave() {
+        if (this.isNewSet) {
+          return this.nameValid && this.channelCount;
+        }
+        return this.getChannelSetIsDirty(this.channelSetId) && this.getChannelSetIsValid(this.channelSetId);
+      },
+      saveButtonTitle() {
+        if (this.saving) {
+          return this.$tr('saving');
+        } else if (!this.isValid) {
+          return this.$tr('invalidCollection');
+        } else if (!this.changed) {
+          return this.$tr('noChangesTitle');
+        } else return this.$tr('saveButtonLabel');
+      },
+      selectableChannels() {
+        return sortBy(this.availableChannels.filter(channel => channel.published), [
+          channel => !this.channels.includes(channel.id),
+          'name',
+        ]);
+      },
+      unpublishedChannels() {
+        return sortBy(this.availableChannels, 'name');
+      },
+      charsLeft() {
+        return this.charLimit - this.description.length;
+      },
+      charLimit() {
+        return 400;
+      },
+      token() {
+        let token = this.channelSet.secret_token;
+        return token && token.display_token;
+      },
+      channelSet() {
+        return this.getChannelSet(this.channelSetId) || {};
+      },
+      channelCount() {
+        return this.channels.length;
+      },
+      isNewSet() {
+        return isDummyId(this.channelSetId)
+      }
+    },
+    methods: {
+      ...mapActions('channelList', ['loadAllChannels']),
+      ...mapActions('channelSet', ['saveChannelSet']),
+      ...mapMutations('channelSet', {
+        updateChannelSet: 'UPDATE_CHANNELSET',
+      }),
+      nameValid(name) {
+        return name && name.length > 0 ? true : this.$tr('titleRequiredText');
+      },
+      loadChannels() {
+        this.loadingChannels = true;
+        this.loadAllChannels().then(() => {
+          this.loadingChannels = false;
+        });
+      },
+      close() {
+        this.$router.push({ name: RouterNames.CHANNEL_SETS });
+      },
+      save() {
+        this.saving = true;
+        return this.saveChannelSet(this.channelSetId).then(newId => {
```

Comment:
When we are saving a new channel set, the action will resolve with the id of the channel set by the server. We navigate to this new id, so that we are away from the dummy id. Using a replace so that if the user closes the modal and then presses back, they are taken to their newly saved channel set.